lib/gds_api/test_helpers/email_alert_api.rb

```diff
@@ -45,26 +45,25 @@ def stub_email_alert_api_does_not_have_subscriber_subscriptions(id)
           .to_return(status: 404)
       end
 
-      def stub_email_alert_api_has_subscription(
-        id,
-        frequency,
-        title: "Some title",
-        subscriber_id: 1,
-        subscriber_list_id: 1000,
-        ended: false
-      )
+      def stub_email_alert_api_has_subscription(id, frequency, optional_overrides = {})
+        response = get_subscription_response(
+          id,
+          fill_gaps_with_default_values(optional_overrides).merge(frequency: frequency)
+        ).to_json
+
         stub_request(:get, "#{EMAIL_ALERT_API_ENDPOINT}/subscriptions/#{id}")
-          .to_return(
-            status: 200,
-            body: get_subscription_response(
-              id,
-              frequency: frequency,
-              title: title,
-              subscriber_id: subscriber_id,
-              subscriber_list_id: subscriber_list_id,
-              ended: ended,
-            ).to_json,
-          )
+          .to_return(status: 200, body: response)
+        stub_request(:get, "#{EMAIL_ALERT_API_ENDPOINT}/subscriptions/#{id}/latest")
```

Comment:
Should this be in a separate method? i.e. `stub_email_alert_api_has_latest_subscription`?

Reply:
That crossed my mind, but _every_ subscription should have a valid response if you call `/latest`. So it would mean otherwise writing this in our tests:

```ruby
stub_email_alert_api_has_subscription(1, 'weekly', :ended)
stub_email_alert_api_has_latest_subscription(1, 'weekly', :ended)
stub_email_alert_api_has_subscription(2, 'weekly')
stub_email_alert_api_has_latest_subscription(2, 'weekly')
```

I think this makes things less readable - it can be misinterpreted as "this subscription should be the latest one", which is not what we want. What we want, if anything, is "this subscription should be queryable against the '/latest' endpoint". But if you've stubbed the email API for a subscription, that should just be done for you IMO.

What do you think?

Reply:
Can you add this explanation to the commit note?